### DISCUSSION TITLE
Is there a way we could download models like unitable from GitHub and pass it as an argument?

### DISCUSSION THREAD

**Discussion by littlebuddha16** [Filimoa/open-parse#59] [Q&A]
openparse-download doesn't work in my firm's VM. I can download the models via GitHub. Could someone help me out on how to pass downloaded model as an argument?

**Comment by Filimoa** [comment]
You can manually just place them in `openparse/weights/unitable`.

```
openparse/
├── processing/
├── tables/
├── text/
└── weights/
    └── unitable/
        ├── unitable_large_bbox.pt
        ├── unitable_large_content.pt
        ├── unitable_large_structure.pt
        ├── vocab_bbox.json
        ├── vocab_cell_6k.json
        └── vocab_html.json
```

**Reply by littlebuddha16** [reply]
Great! Could you also tell me the place I could download the above weights?

**Reply by Filimoa** [reply]
```
urls = [
        "https://huggingface.co/poloclub/UniTable/resolve/main/unitable_large_structure.pt",
        "https://huggingface.co/poloclub/UniTable/resolve/main/unitable_large_bbox.pt",
        "https://huggingface.co/poloclub/UniTable/resolve/main/unitable_large_content.pt",
        "https://sergey-filimonov.nyc3.digitaloceanspaces.com/open-parse/weights/vocab_bbox.json",
        "https://sergey-filimonov.nyc3.digitaloceanspaces.com/open-parse/weights/vocab_cell_6k.json", 
        "https://sergey-filimonov.nyc3.digitaloceanspaces.com/open-parse/weights/vocab_html.json",  
    ]
```

**Reply by littlebuddha16** [reply]
I'm getting below error; Could you also give me the links to download the below model and the directory location to place it?
`OSError: We couldn't connect to 'https://huggingface.co/' to load this file, couldn't find it in the cached files and it looks like microsoft/table-transformer-detection is not the path to a directory containing a file named config.json.
Checkout your internet connection or see how to run the library in offline mode at 'https://huggingface.co/docs/transformers/installation#offline-mode'.`

**Reply by Filimoa** [reply]
Looks like a corporate firewall issue that's beyond the scope of this library. Good luck.

**Reply by littlebuddha16** [reply]
Can't we download the model like unitable and place it in the openparse directory?

**Reply by Filimoa** [reply]
This is tied to the table detection algorithm which is downloaded from hugging face.  Not exactly sure how it downloads these - https://huggingface.co/docs/transformers/en/model_doc/table-transformer.
